.github/workflows/_cleanup.yml

```diff
@@ -0,0 +1,50 @@
+name: Setup Workload Identity 
+
+on:
+  workflow_call:
+    inputs:
+      cluster-name: 
+        required: true
+        type: string
+        description: Azure AKS Cluster Name
+  workflow_dispatch:
+    inputs:
+       cluster-name:
+        required: true
+        type: string
+        description: Azure AKS Cluster Name
+
+
+jobs:
+  setup-workload-identity:
+    name: Setup Workload Identity 
+    runs-on: ubuntu-latest
+    steps:
+
+    #   - name: Log into Azure
+    #     run: |
+    #       az login --service-principal \
+    #         --username ${{ secrets.AZURE_SERVICE_PRINCIPAL_APP_ID }} \
+    #         --password ${{ secrets.AZURE_SERVICE_PRINCIPAL_PASSWORD }} \
+    #         --tenant ${{ secrets.AZURE_SERVICE_PRINCIPAL_TENANT }}
+            
+      - name: Install Dependencies 
+        id: install-dependencies 
+        run: |
+
+            curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
+            chmod +x kubectl
+            sudo mv kubectl /usr/local/bin
+            kubectl version 2>&1 || true
+
+      - name: Cleanup
+        id: cleanup
+        run: | 
+          
+          kubectl delete pod kafka-producer 
```

Comment:
For cleanup, I would just delete the whole cluster/node since that should delete all the resources under it.

Reply:
Yes, the cluster gets removed after the test is done. This was not pushed. I'll send you a note when the PR is ready for review.